File: app/src/main/java/com/ipd/taxiu/adapter/FriendListAdapter.java
package com.ipd.taxiu.adapter;

import android.content.Context;
import android.content.Intent;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.RelativeLayout;

import com.ipd.taxiu.R;
import com.ipd.taxiu.bean.FriendBean;
import com.ipd.taxiu.bean.FriendListBean;
import com.ipd.taxiu.ui.activity.referral.HomepageActivity;

import java.util.List;

/**
 * Created by Miss on 2018/7/25
 * 好友列表adapter
 */
public class FriendListAdapter extends RecyclerView.Adapter<FriendListAdapter.ViewHolder> {
    private Context context;
    private List<FriendBean> datas;

    public FriendListAdapter(Context context, List<FriendBean> datas) {
        this.context = context;
        this.datas = datas;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        return new FriendListAdapter.ViewHolder(LayoutInflater.from(context).inflate(R.layout.item_friend_list, parent, false));
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        holder.rl_friend_list.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(context, HomepageActivity.class);
                context.startActivity(intent);
            }
        });
    }

    @Override
    public int getItemCount() {
        return datas.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        RelativeLayout rl_friend_list;
        public ViewHolder(View itemView) {
            super(itemView);
            rl_friend_list = itemView.findViewById(R.id.rl_friend_list);
        }
    }
}

File: app/src/main/java/com/ipd/taxiu/adapter/FriendListAdapter.kt
package com.ipd.taxiu.adapter

import android.app.Activity
import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.ipd.taxiu.R
import com.ipd.taxiu.bean.AttentionBean
import com.ipd.taxiu.imageload.ImageLoader
import com.ipd.taxiu.platform.http.HttpUrl.IMAGE_URL
import com.ipd.taxiu.ui.activity.referral.HomepageActivity
import kotlinx.android.synthetic.main.item_friend_list.view.*

/**
Created by Miss on 2018/8/13
 */
class FriendListAdapter(val context: Context, private val data: List<AttentionBean>) : RecyclerView.Adapter<FriendListAdapter.ViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): ViewHolder {
        return ViewHolder(LayoutInflater.from(context).inflate(R.layout.item_friend_list, parent, false))
    }

    override fun getItemCount(): Int = data?.size ?: 0
    override fun onBindViewHolder(holder: ViewHolder?, position: Int) {
        ImageLoader.loadAvatar(context, IMAGE_URL + data[position].LOGO, holder?.itemView?.civ_friend_header)
        holder?.itemView?.tv_friend_nickname?.text = data[position].NICKNAME
        holder?.itemView?.tv_create_time?.text = "注册日期：" + data[position].CREATETIME
        holder?.itemView?.setOnClickListener {
            HomepageActivity.launch(context as Activity, data[position].USER_ID)
        }
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView)
}